boot/unlocked_state.go

```diff
@@ -0,0 +1,95 @@
+package boot
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+)
+
+// states for partition state
+const (
+	// states for LocateState
+	PartitionFound      = "found"
+	PartitionNotFound   = "not-found"
+	PartitionErrFinding = "error-finding"
+	// states for MountState
+	PartitionMounted          = "mounted"
+	PartitionErrMounting      = "error-mounting"
+	PartitionAbsentOptional   = "absent-but-optional"
+	PartitionMountedUntrusted = "mounted-untrusted"
+	// states for UnlockState
+	PartitionUnlocked     = "unlocked"
+	PartitionErrUnlocking = "error-unlocking"
```

Comment:
Any reason for not using typed enums instead of strings?

Reply:
I did not want to extract and rewrite everything. I am not sure why they were not enums in the first place. I will add a FIXME.

Reply:
because kernel snapd version and snapd version might be out of sync, it's much easier to mess up int enums than strings, I don't think we want a FIXME, but maybe I'm misunderstanding the plan here

Reply:
I will remove the TODO then.

Reply:
Sorry for confusion, I meant typed string values not ints

```golang

type PartitionLocateState string

const (
	// states for LocateState
	PartitionFound         PartitionLocateState = "found"
	PartitionNotFound      PartitionLocateState = "not-found"
	PartitionErrFinding    PartitionLocateState = "error-finding"
)
```